src/api/PluginManager.ts

```diff
@@ -11,22 +12,35 @@ import { Logger } from "../utils/Logger";
 
 const logger = new Logger("PluginManager");
 export const plugins = {} as Record<string, Plugin>;
+export const disabledPlugins = {} as Record<string, PluginManifest>;
 
 export function isPluginEnabled(plugin: string) {
     return window.Aliucord.settings.get("plugins", {})[plugin] !== false;
 }
 
 export function enablePlugin(plugin: string) {
     const plugins = window.Aliucord.settings.get("plugins", {});
+    if (plugins[plugin] === true) throw new Error(`Plugin ${plugin} is already enabled.`);
+    delete disabledPlugins[plugin];
     plugins[plugin] = true;
     window.Aliucord.settings.set("plugins", plugins);
+
+    const bundleZip = new ZipFile(PLUGINS_DIRECTORY + `${plugin}.zip`, 0, "r");
+    const pluginBuffer = loadPluginBundle(bundleZip);
+
+    const pluginClass = AliuHermes.run(plugin, pluginBuffer) as typeof Plugin;
+    plugins[plugin] = new pluginClass(plugins[plugin].manifest).start();
+
     Toasts.open({ content: `Enabled ${plugin}`, source: getAssetId("Check") });
     logger.info(`Enabled plugin: ${plugin}`);
 }
 
 export function disablePlugin(plugin: string) {
     const plugins = window.Aliucord.settings.get("plugins", {});
-    plugins[plugin] = false;
+    disabledPlugins[plugin] = window.Aliucord.pluginManager.plugins[plugin].manifest;
+    window.Aliucord.pluginManager.plugins[plugin].stop();
```

Comment:
why does this use it via the window

Reply:
because `plugins` is already getting const'd above it (idk why, not done by me)

Reply:
rename it then